hoot-services/src/main/java/hoot/services/controllers/grail/WaitOsmUpdate.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2020 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+package hoot.services.controllers.grail;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.time.LocalDateTime;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+import hoot.services.command.CommandResult;
+import hoot.services.utils.DbUtils;
+
+import static hoot.services.HootProperties.PRIVATE_OVERPASS_URL;
+import static hoot.services.HootProperties.PUBLIC_OVERPASS_URL;
+import static hoot.services.HootProperties.replaceSensitiveData;
+
+
+/**
+ * Used for pushing OSM data to the database
+ */
+class WaitOsmUpdate extends GrailCommand {
+    private static final Logger logger = LoggerFactory.getLogger(WaitOsmUpdate.class);
+    private String jobId;
+    private final Class<?> caller;
+    private final int timeoutTime = 300000;
+    private final int sleepTime = 30000;
+
+    WaitOsmUpdate(String jobId, GrailParams params, String debugLevel, Class<?> caller) {
+        super(jobId, params);
+        logger.info("Params: " + params);
+
+        this.jobId = jobId;
+        this.caller = caller;
+    }
+
+    @Override
+    public CommandResult execute() {
+        CommandResult commandResultStart = new CommandResult();
+        commandResultStart.setStart(LocalDateTime.now());
+        commandResultStart.setCommand("OSM sync wait");
+        commandResultStart.setJobId(jobId);
+        commandResultStart.setStdout("Starting wait on osm sync...\n");
+        commandResultStart.setStderr("");
+        commandResultStart.setPercentProgress(0);
+        commandResultStart.setCaller(caller.getName());
+        commandResultStart.setExitCode(CommandResult.SUCCESS);
+        commandResultStart.setFinish(LocalDateTime.now());
+        DbUtils.upsertCommandStatus(commandResultStart);
+        DbUtils.completeCommandStatus(commandResultStart);
+
+        if (GrailResource.isPrivateOverpassActive()) {
+            long timeSpent = 0;
+            String url = null;
+
+            long lastId = DbUtils.getLastPushedId(jobId);
+
+            String query = "[out:json];"
+                    + "("
+                    + "node(id:" + lastId + ");"
+                    + "way(id:" + lastId + ");"
+                    + "rel(id:" + lastId + ");"
+                    + ");"
+                    + "out tags;";
```

Comment:
I think the `lastId` is a changeset id not a feature id.  I think we might need to use the `newer` filter using the starting polling datetime.
https://wiki.openstreetmap.org/wiki/Overpass_API/Overpass_QL#newer
The query will return `meta` that will include changeset id for the returned features.